### ISSUE TITLE
sql: CREATE OR REPLACE VIEW

### ISSUE THREAD

**Issue by godofdream** [cockroachdb/cockroach#24897]
**FEATURE REQUEST**
Postgresql supports CREATE OR REPLACE VIEW 
(https://www.postgresql.org/docs/10/static/sql-createview.html)
Postgres limitates the replace view function to only add new columns.

CockroachDB could support this sqlstatement too, as it's very convenient way to replace views.
A Workaround would be dropping and creating a view in a transaction. (currently blocked by #12123)

Importance: Nice-to-have postgresql compatibility

**Comment by danhhz** [comment]
Assigning to @knz for triage

**Comment by knz** [comment]
@godofdream thanks for your interest in CockroachDB.

Do you have an example app or framework in mind that already uses this feature? Where compatibility is blocked because CockroachDB does not support it?

**Comment by godofdream** [comment]
@knz Currently we use Postgres with the replace view function in a closed-source project. We use the replace as function for migrating to newer versions of the view. It's syntactic sugar as it could be done with a transaction alternativly.

Besides our usecase i didn't find any opensource projects using the create or replace view feature of postgres.

@a-robinson marked CREATE OR REPLACE as TODO in the create_view code
[https://github.com/cockroachdb/cockroach/blob/36e3bc6a469c3abc30d3ee53f70fd0a146432608/pkg/sql/create_view.go#L122]

**Comment by knz** [comment]
@godofdream thanks for your explanation. I wanted to hear the context of your request so that we know how to prioritize this appropriately.

**Comment by BramGruneir** [comment]
@awoods187 This seems like a fairly straightforward ask.  Can we get this into 20.1?

**Comment by knz** [comment]
(Not completely straighforward, we still have to check the type compatibility, and update all the references.)

**Comment by BramGruneir** [comment]
Yes, but that would just follow the `alter view` command that already works.

**Comment by knz** [comment]
oh I did not know that alter view was already a thing.

**Comment by BramGruneir** [comment]
Ah, we only support rename in alter view :(  Let's get that fixed for 20.1 first.

**Comment by RoachietheSupportRoach** [comment]
Zendesk ticket #[3675](https://cockroachdb.zendesk.com/agent/tickets/3675) has been linked to this issue.

**Comment by lukaseder** [comment]
Folks, is there a way to see from an issue like this in what version the feature will ship?

**Comment by knz** [comment]
1. find the PR number that addresses the issue
2. check out each `release-XX` branch in turn
3. in each branch, search for that PR number in the git log

If a branch has the PR, then that release has the fix.

**Comment by knz** [comment]
FYI in this particular case the first release to have it will be v20.2

**Comment by lukaseder** [comment]
Great, thanks a lot, @knz
